### PR TITLE
Bit 57 local training

### PR DESCRIPTION
**Pull request by Eugene-hu**
Networking update: subscription works with both ipv4 and ipv6 addresses

Miner update: updated template learning parameters, introduced a momentum hyperparameter, and introduced a learning scheduler.  Removed some old debugging print statements

### CODE REVIEW

File: miners/text/sample_config.txt [hunk 1]
```diff
@@ -14,14 +14,14 @@ dendrite.max_worker_threads: 150
 dendrite.requires_grad: true
 dendrite.timeout: 5
 
-logging.debug: false
+logging.debug: true
```
Comment (shibshib) [inline miners/text/sample_config.txt:17]: Should logging debug be true by default?

File: miners/text/sample_config.txt [hunk 2]
```diff
@@ -14,14 +14,14 @@ dendrite.max_worker_threads: 150
...
 logging.logging_dir: ~/.bittensor/miners/
 logging.record_log: false
 logging.trace: false
 
 miner.device: cpu
-miner.epoch_length: 500
-miner.learning_rate: 0.03
+miner.epoch_length: 100
+miner.learning_rate: 1
```
Comment (shibshib) [inline miners/text/sample_config.txt:24]: learning rate 1? isn't that epically high? :D

File: miners/text/template_miner.py [hunk 3]
```diff
@@ -82,9 +82,9 @@ def add_args( parser: argparse.ArgumentParser ):
     def init_weights(self):
         initrange = 0.1
         self.remote_decoder.weight.data.uniform_(-initrange, initrange)
-        self.remote_decoder.bias.data.zero_()
+        #self.remote_decoder.bias.data.zero_()
```
Comment (shibshib) [inline miners/text/template_miner.py]: If we're not using this, let's just remove it altogether.

**Review by shibshib — APPROVED** [review]
Generally looks good, just a couple nits and comments

File: miners/text/template_miner.py [hunk 4]
```diff
@@ -262,8 +264,9 @@ def config() -> 'bittensor.Config':
         # ---- Add miner args.
         parser = argparse.ArgumentParser()
         parser.add_argument('--miner.config', type=str, help='If set, defaults are overridden by passed file.')
-        parser.add_argument('--miner.learning_rate', type=float, help='Training initial learning rate.', default=3e-2)
+        parser.add_argument('--miner.learning_rate', type=float, help='Training initial learning rate.', default=1)
```
Comment (ivanezeigbo) [inline miners/text/template_miner.py:268]: just fyi if we decide that the learning rate of 1 is high

**Review by ivanezeigbo — APPROVED** [review]
Just left some comments

**Review by isabella618033 — APPROVED** [review]

File: miners/text/sample_config.txt (hunk 1 above)
Comment (Eugene-hu) [reply miners/text/sample_config.txt:17]: logging should be turned off by default, but I left it on due to the current bug where it doesn't display any of the subscription logs

File: miners/text/sample_config.txt (hunk 2 above)
Comment (Eugene-hu) [reply miners/text/sample_config.txt:24]: yeap, it is extremely high. I actually tuned it down from 5 in the original PyTorch example

File: miners/text/template_miner.py (hunk 3 above)
Comment (Eugene-hu) [reply miners/text/template_miner.py]: Shows good, will remove

File: miners/text/sample_config.txt [hunk 5]
```diff
@@ -14,14 +14,14 @@ dendrite.max_worker_threads: 150
...
 miner.name: gpt2_exodus
```
[the remark this answers is not in the document]
Comment (Eugene-hu) [reply miners/text/sample_config.txt]: removed!

**Review by unconst — APPROVED** [review]
